### PR TITLE
Update pre-commit

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Redo #14987.

### Does your PR introduce any breaking changes? If yes, please list them.
None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [n/a] Did you verify new and **existing tests pass** locally with your changes?
- [n/a] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @carmocca @akihironitta @borda

### CODE REVIEW

File: src/pytorch_lightning/accelerators/accelerator.py [hunk 1]
```diff
@@ -29,9 +29,10 @@ class Accelerator(_Accelerator, ABC):
     """
 
     def setup_environment(self, root_device: torch.device) -> None:
-        """
-        .. deprecated:: v1.8.0
-            This hook was deprecated in v1.8.0 and will be removed in v1.10.0. Please use ``setup_device()`` instead.
+        """..
+
+        deprecated:: v1.8.0
```
Comment (akihironitta) [inline src/pytorch_lightning/accelerators/accelerator.py:34]: Note to self: find out a way to avoid this change

Comment (Borda) [reply src/pytorch_lightning/accelerators/accelerator.py:34]: Yes, you just need to write any text just before it...

**Comment by akihironitta** [comment]
It turned out docformatter doesn't do well on the sphinx directive: https://github.com/PyCQA/docformatter/issues/78

```diff
- .. deprecated:: v1.8.0
-     This hook was deprecated in v1.8.0 and will be removed in v1.10.0. Please use
-     ``setup_device()`` instead.
+ .. deprecated:: v1.8.0 This hook was deprecated in v1.8.0 and will be removed in v1.10.0. Please use
+ ``setup_device()`` instead.
```

File: src/pytorch_lightning/profiler/base.py [hunk 2]
```diff
@@ -52,7 +52,10 @@ def teardown(self, **kwargs: Any) -> None:
 
 
 class BaseProfiler(Profiler):
-    """
+    """If you wish to write a custom profiler, you should inherit from this class.
+
+    Note that this is deprecated.
+
```
Comment (Borda) [inline src/pytorch_lightning/profiler/base.py:58]:
```suggestion
```
I think this is not needed, but let's try...

File: src/pytorch_lightning/profiler/profiler.py [hunk 3]
```diff
@@ -16,7 +16,10 @@
 
 
 class Profiler(NewProfiler):
-    """
+    """If you wish to write a custom profiler, you should inherit from this class.
+
+    Note that this is deprecated.
+
```
Comment (Borda) [inline src/pytorch_lightning/profiler/profiler.py:22]:
```suggestion
```

File: src/pytorch_lightning/strategies/deepspeed.py [hunk 4]
```diff
@@ -68,7 +68,11 @@ def remove_module_hooks(model: torch.nn.Module) -> None:
 
 
 class LightningDeepSpeedModule(_LightningModuleWrapperBase):
-    """
+    """Wraps the user's LightningModule and redirects the forward call to the appropriate method, either
+    ``training_step``, ``validation_step``, ``test_step``, or ``predict_step``
+
+    Note that this is deprecated.
+
```
Comment (Borda) [inline src/pytorch_lightning/strategies/deepspeed.py:75]:
```suggestion
```

**Review by Borda — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Comment by carmocca** [comment]
Docs failures look real

**Comment by akihironitta** [comment]
Let me finish this tomorrow (as it's a holiday tomorrow) or this weekend.

**Comment by Borda** [comment]
> Let me finish this tomorrow (as it's a holiday tomorrow) or this weekend.

no rush, just checking :)

**Comment by akihironitta** [comment]
Closing this as I currently have no bandwith to finish this PR...

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -75,7 +76,7 @@ repos:
         exclude: docs/source-app
 
   - repo: https://github.com/psf/black
-    rev: 22.6.0
+    rev: 22.10.0
```
Comment (carmocca) [inline .pre-commit-config.yaml:79]: You could update the rest of the hooks and leave black out for now, as that's the one that requires more work.

**Comment by carmocca** [comment]
Opened #16257 so somebody finishes the docformatter upgrade. The other actions will be upgraded in https://github.com/Lightning-AI/lightning/pull/16224
